feat(dropdown): initial design guidance (#25)
vmware-clarity/clarity-guidance: 1011-dropdown-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-21
---

> # Dropdown Component

Reviewer: ```suggestion
# Dropdown Design Guidance
```

> [Dropdown](https://clarity.design/documentation/dropdown) component allows the user to choose an option or action from a contextual list. The source of the dropdown is usually a button.

Reviewer: ```suggestion
The [dropdown](https://clarity.design/documentation/dropdown) component allows the user to choose an option or action from a contextual list. The trigger of the dropdown is usually a button.
```

## Guidance

> - **Must** be used to take an immediate action or navigate the user outside of the current context.

Reviewer 2: ```suggestion
- Design teams **must** use a dropdown to facilitate immediate actions or navigate users outside of the current context, ensuring a seamless transition and clear user flow.
```

> - **Must** NOT use a dropdown menu to hide primary actions. They must never be hidden behind a dropdown menu.

Reviewer: ```suggestion
- **Must not** use a dropdown menu to hide primary actions. Such actions **must not** be hidden behind a dropdown menu.
```

> - **Must not** use a dropdown menu to hide primary actions. Such actions **must not** be hidden behind a dropdown menu.

Reviewer 2: ```suggestion
- Design teams **must not** use a dropdown menu to hide primary actions, as these actions should be readily accessible and not concealed behind a dropdown.
```

> - **Should** use headers and dividers for longer list of items  

Reviewer 2: ```suggestion
- Design teams **must not** use a dropdown menu to hide primary actions, as these actions should be readily accessible and not concealed behind a dropdown. 
```

> - **Should** use nested dropdown for organizing longer list into categories that the user can click to reveal deeper levels.

Reviewer 2: ```suggestion
- Design teams **should** use nested dropdowns to organize longer lists into categories, allowing users to click to reveal deeper levels and enhancing navigation efficiency.
```

> - **Should** NOT use dropdown menu as a form input. Use the `select` component instead.

Reviewer: ```suggestion
- **Must not** use dropdown menu as a form control. Use a [select](https://clarity.design/documentation/select) instead.
```

This one isn't optional. A dropdown *cannot* be used as form control because the items **must** for immediate actions or navigation.

> - **Must not** use dropdown menu as a form control. Use a [select](https://clarity.design/documentation/select) instead.

Reviewer 2: ```suggestion
- Design teams **must not** use a dropdown menu as a form control for user input. Instead, they should use a [select](https://clarity.design/documentation/select) component, which is better suited for handling form data and allows for clearer interaction and selection.
```

> - Its best practice to limit Nested dropdown menus to three levels.

Reviewer: ```suggestion
- Nested dropdown menus *should* be limited to three levels.
```

Let's use our keywords. :)

> - Nested dropdown menus *should* be limited to three levels.

Reviewer 2: ```suggestion
- Design teams **should** limit nested dropdown menus to three levels to prevent confusion and maintain clarity, ensuring users can easily navigate without feeling overwhelmed.
```

## Content Guidelines

> - **Must** keep the description of the dropdown options as short as possible to improve readability.

Reviewer 2: ```suggestion
- Design teams **must** keep the descriptions of dropdown options as short as possible to enhance readability and ensure users can quickly understand their choices.
```

> - Phrase the label that clearly indicate the `dropdown` action.

Reviewer: ```suggestion
- Phrase the label that clearly indicate the dropdown action.
```

No need for inline code styling here.

> - Phrase the label that clearly indicate the dropdown action.

Reviewer: ```suggestion
- **Should** phrase the label that clearly indicate the dropdown action.
```

> - **Should** phrase the label that clearly indicate the dropdown action.

Reviewer 2: ```suggestion
- Design teams **should** phrase the label to clearly indicate the dropdown action, helping users understand what to expect when interacting with the dropdown.
```
